src/commonMain/kotlin/io/github/detekt/sarif4k/SarifDataBindings.kt

```diff
@@ -339,13 +344,26 @@ data class Address (
  *
  * Key/value pairs that provide additional information about the version control details.
  */
-@Serializable
-data class PropertyBag (
+@Serializable(with = PropertyBagSerializer::class)
+@JvmInline
+value class PropertyBag(private val value: Map<String, Any?>) : Map<String, Any?> by value {
     /**
      * A set of distinct strings that provide additional information.
      */
-    val tags: List<String>? = null
-)
+    val tags: List<String>? get() = (value["tags"] as Collection<*>?)?.map { it as String }
+}
+
+object PropertyBagSerializer : KSerializer<PropertyBag> {
```

Comment:
Does this have to be public?

Reply:
Good point, no, probably not. I'll change.

Reply:
How can we test to see if it works when actually used as a library?
(It might work in tests because of "friend rules".)

Reply:
I changed this code. Now it's a companion object to follow the same pattern than the other `KSerializer` of this file. And those `companion object` are all `public`. We could change that, but we should do that in other PR.